src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1887,19 +1962,23 @@ const PRIMITIVES: &[(&str, Res)] = &[
     ("f64", Res::PrimTy(hir::PrimTy::Float(rustc_ast::FloatTy::F64))),
     ("str", Res::PrimTy(hir::PrimTy::Str)),
     ("bool", Res::PrimTy(hir::PrimTy::Bool)),
-    ("true", Res::PrimTy(hir::PrimTy::Bool)),
-    ("false", Res::PrimTy(hir::PrimTy::Bool)),
```

Comment:
I'm not entirely certain if removing these from `is_primitive` is correct, but the tests don't complain. True and false are handled separately in `is_bool_value`